### PR TITLE
Updates policy to better warn players for any existing antagonist policy

### PR DESCRIPTION
**Pull request by MrMelbert**
# Don't merge this without leading the headmins know so they can update the server configs

### For headmins: 

The following roles need to be updated:
- "Obsession" -> `/datum/antagonist/obessed`
- "Heretic" -> `/datum/antagonist/heretic` (Though this one will still work, change it to message them on spawn)
- "Headslug Changeling" -> `/datum/antagonist/changeling/headslug`
- "Pyroclastic Anomaly Slime" -> `/datum/antagonist/pyro_slime` (Though this one will still work, technically, change it to message them on spawn)

## About The Pull Request

So, we have this whole system in the code to allow all mob typepaths, species typepaths, and even antag datum typepaths to have a set policy

But we like, don't use it? For some reason we rather set a million `get_policy` calls than use this. They're all stuck in the policy verb and not sent to the player in any obvious. 

(Also, it's super fragile, which I hate. Change the typepath and the policy silently fails. Maybe someone can change that afterwards)

This PR updates that. All antag datums will check in `on_gain` for policy to show to the player if there's any. 

It doesn't solve the entire issue. Mob policy and species policy is not conveyed to the player in any way BESIDES the policy verb which no one uses. So people have no idea that it's a thing. But it helps a bit. 

## Why It's Good For The Game

Better ways to convey to players how they should be playing. 

Also Valentines day, because... it's crazy. 

## Changelog

:cl: Melbert
config: Policy.json config got updated. The following values need to be updated:
config: "Obsession" -> /datum/antagonist/obessed
config: "Heretic" -> /datum/antagonist/heretic
config: "Headslug Changeling" -> /datum/antagonist/changeling/headslug
config: "Pyroclastic Anomaly Slime" -> /datum/antagonist/pyro_slime
config: ALL antags in the policy.json file will now get notified of any existing policy set by the admin team, rather than only a handful.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -543,13 +540,12 @@ SUBSYSTEM_DEF(job)
 
 	if(player_client)
 		if(job.req_admin_notify)
-			to_chat(player_client, "<span class='infoplain'><b>You are playing a job that is important for Game Progression. If you have to disconnect, please notify the admins via adminhelp.</b></span>")
+			to_chat(player_client, "<span class='infoplain'><b>You are playing a job that is important for Game Progression. \
+				If you have to disconnect, please notify the admins via adminhelp.</b></span>")
```
Comment (timothymtorres) [inline code/controllers/subsystem/job.dm:544]:
```suggestion
			to_chat(player_client, span_infoplain("<b>You are playing a job that is important for Game Progression. \
				If you have to disconnect, please notify the admins via adminhelp.</b>"))
```

Not sure if `<b>` works with with the spanclass helper but I'm going to point it out.

File: code/datums/mind/_mind.dm [hunk 2]
```diff
@@ -499,6 +499,19 @@
...
 
+/// Sets us to the passed job datum, then greets them to their new job.
+/// Use this one for when you're assigning this mind to a new job for the first time,
+/// or for when someone's recieving a job they'd really want to be greeted to.
+/datum/mind/proc/set_assigned_role_with_greeting(datum/job/new_role)
+	. = set_assigned_role(new_role)
+	if(!.)
+		return
+
+	to_chat(src, "<span class='infoplain'><b>You are the [new_role.title].</b></span>")
```
Comment (timothymtorres) [inline code/datums/mind/_mind.dm:510]:
```suggestion
	to_chat(src, span_infoplain("<b>You are the [new_role.title].</b>"))
```

Same.

**Comment by MrStonedOne** [comment]
> # Don't merge this without leading the headmins know so they can update the server configs
> ### For headmins:
> The following roles need to be updated:
> 
> * "Obsession" -> `/datum/antagonist/obessed`
> * "Heretic" -> `/datum/antagonist/heretic` (Though this one will still work, change it to message them on spawn)
> * "Headslug Changeling" -> `/datum/antagonist/changeling/headslug`
> * "Pyroclastic Anomaly Slime" -> `/datum/antagonist/pyro_slime` (Though this one will still work, technically, change it to message them on spawn)

This information has to be placed in the config changelog entry, not just here in the pr.

**Comment by san7890** [comment]
i'm headmins

**Comment by san7890** [comment]
just now realizing you mispelled `obsessed` in both the PR body and changelog... it's okay the config is okay because i spotted that

**Pull request merged**